components/form-builder/app/Nagware.tsx

```diff
@@ -0,0 +1,106 @@
+import React from "react";
+import { useTranslation } from "next-i18next";
+import { Attention } from "@components/globals/Attention/Attention";
+import { NagLevel, NagwareResult } from "@lib/nagware";
+
+export const Nagware = ({ nagwareResult }: { nagwareResult: NagwareResult }) => {
+  const { t } = useTranslation("form-builder");
+
+  if (nagwareResult.level === NagLevel.None) return <></>;
+
+  const generateNotificationContent = (): {
```

Comment:
I'd suggest maybe using a [useMemo](https://react.dev/reference/react/useMemo) function here to memoize the `generateNotificationContent`.  This way the function isn't recalculated on every render and will only recalculate is one of the listed dependencies change.